pkg/store/event/event.go

```diff
@@ -262,44 +262,65 @@ func (e *eventStore) DeleteIncompleteEvents(_ context.Context) error {
 	var resList []*core.Event
 	var err error
-	var startTime time.Time
-
-	if podName != "" {
-		resList, err = e.ListByPod(context.Background(), podNamespace, podName)
-	} else if podNamespace != "" {
-		resList, err = e.ListByNamespace(context.Background(), podNamespace)
-	} else if uid != "" {
-		resList, err = e.ListByUID(context.Background(), uid)
-	} else {
-		resList, err = e.List(context.Background())
+	var startTimeBegin, finishTimeEnd time.Time
+
+	if filter.StartTimeBeginStr != "" {
+		startTimeBegin, err = time.Parse(time.RFC3339, strings.Replace(filter.StartTimeBeginStr, " ", "+", -1))
+		if err != nil {
+			return nil, fmt.Errorf("the format of the startTimeBegin is wrong")
+		}
 	}
-	if err != nil {
-		return resList, err
+	if filter.FinishTimeEndStr != "" {
+		finishTimeEnd, err = time.Parse(time.RFC3339, strings.Replace(filter.FinishTimeEndStr, " ", "+", -1))
+		if err != nil {
+			return nil, fmt.Errorf("the format of the finishTimeEnd is wrong")
+		}
 	}
 
-	if startTimeStr != "" {
-		startTime, err = time.Parse(time.RFC3339, strings.Replace(startTimeStr, " ", "+", -1))
-		if err != nil {
-			return nil, fmt.Errorf("the format of the time is wrong")
+	if filter.PodName != "" {
+		resList, err = e.ListByPod(context.Background(), filter.PodNamespace, filter.PodName)
+	} else if filter.PodNamespace != "" {
+		resList, err = e.ListByNamespace(context.Background(), filter.PodNamespace)
+	} else {
+		query, args := constructQueryArgs(filter.ExperimentName, filter.ExperimentNamespace, filter.UID, filter.Kind, filter.StartTimeBeginStr, filter.FinishTimeEndStr)
+		// List all events
+		if len(args) == 0 {
+			if err := e.db.Model(core.Event{}).Find(&resList).Error; err != nil && !gorm.IsRecordNotFoundError(err) {
+				return resList, err
+			}
+		} else {
+			if err := e.db.Where(query, args...).Find(&resList).Error; err != nil &&
+				!gorm.IsRecordNotFoundError(err) {
+				return resList, err
+			}
 		}
 	}
+	if err != nil {
+		return resList, err
+	}
 
 	eventList := make([]*core.Event, 0)
 	for _, event := range resList {
-		if experimentName != "" && event.Experiment != experimentName {
+		if filter.ExperimentName != "" && event.Experiment != filter.ExperimentName {
+			continue
+		}
+		if filter.ExperimentNamespace != "" && event.Namespace != filter.ExperimentNamespace {
 			continue
 		}
-		if experimentNamespace != "" && event.Namespace != experimentNamespace {
+		if filter.UID != "" && event.ExperimentID != filter.UID {
 			continue
 		}
-		if uid != "" && event.ExperimentID != uid {
+		if filter.Kind != "" && event.Kind != filter.Kind {
 			continue
 		}
-		if startTimeStr != "" && event.StartTime.Before(startTime) && !event.StartTime.Equal(startTime) {
+		if filter.StartTimeBeginStr != "" && event.StartTime.Before(startTimeBegin) && !event.StartTime.Equal(startTimeBegin) {
+			continue
+		}
+		if filter.FinishTimeEndStr != "" && event.FinishTime.After(finishTimeEnd) && !event.FinishTime.Equal(finishTimeEnd) {
 			continue
 		}
```

Comment:
I think the all the if statements above are not necessary since you are already using `constructQueryArgs` to query from the database.

Reply:
@yeya24
These "if" are still needed, because if we specify "podName" and "podNamespace", then we will call "ListByPod" or "ListByNamespace". In this case, "constructQueryArgs" is not used, so it is not filtered according to other parameters. Therefore, it is necessary to add "if" in the following loop to judge.

Reply:
Check these two situations to judge whether to run these extra `if` checks?

Like:

```go
if filter.PodName != "" || filter. PodNamespace != "" {
        // if ...
        // if ...
}
```

Reply:
I have tested the code and seems there is a bug when comparing event time caused by different timezones.

The `event.FinishTime` is stored in the database with current timezone, for example in my case it is `2020-07-07 09:06:20 -0400 -0400`. But `finishTime` is parsed to UTC time `2020-07-07 09:10:11.826 +0000 UTC`, so during the comparison, event.FinishTime.After(finishTime) returns true and all the events are filtered.